smpp/receiver.go

```diff
@@ -47,7 +48,7 @@ type Receiver struct {
 
 // HandlerFunc is the handler function that a Receiver calls
 // when a new PDU arrives.
-type HandlerFunc func(p pdu.Body)
+type HandlerFunc func(receiverName string, p pdu.Body)
```

Comment:
It would be better to keep this generic. Instead of Name, we could have data of interface{} type. The handlerfunc can decide what it is used for.

Reply:
Changed as suggested